sips/sip-temporary_keypairs-des-128.md

```diff
@@ -0,0 +1,123 @@
+                kp_strings = serde_json::from_str(&*decode_data)
+                    .map_err(|e| anyhow!("Can't deserialize FileBasedKeystore from {:?}: {e}", path))?;
+            }
+2. for fullnode.yaml , network.yaml files, we should change the serder function
+   #[serde(default = "default_authority_key_pair")]
+   #[serde(serialize_with = "serialize_with_aes_encode_authoritykey")]
+   #[serde(deserialize_with = "deserialize_with_aes_encode_authoritykey")]
+   pub protocol_key_pair: AuthorityKeyPairWithPath,
+
+   #[serde(default = "default_key_pair")]
+   #[serde(serialize_with = "serialize_with_aes_encode")]
+   #[serde(deserialize_with = "deserialize_with_aes_encode")]
+   pub worker_key_pair: KeyPairWithPath,
+   #[serde(default = "default_key_pair")]
+   #[serde(serialize_with = "serialize_with_aes_encode")]
+   #[serde(deserialize_with = "deserialize_with_aes_encode")]
+   pub account_key_pair: KeyPairWithPath,
+   #[serde(default = "default_key_pair")]
+   #[serde(serialize_with = "serialize_with_aes_encode")]
+   #[serde(deserialize_with = "deserialize_with_aes_encode")]
+   pub network_key_pair: KeyPairWithPath,
+ 
+  the deserialize_with_aes_encode_xxx is a little complex, for we need to support both string and mapping type,
+  fn deserialize_with_aes_encode_authoritykey<'de, D>(deserializer: D) -> Result<AuthorityKeyPairWithPath, D::Error>
+  where
+  D: Deserializer<'de>,
+  {
+  let value: serde_yaml::Value = match serde_yaml::Value::deserialize(deserializer) {
+  Ok(value) => value,
+  Err(_) => return Err(D::Error::custom("Failed to deserialize value as YAML")),
+  };
+
+    match value {
+        serde_yaml::Value::String(mut s) => {
+            if s.starts_with(DEFAULT_AES_PREFIX) {
+                s = node_des_128_decode(s);
+            }
+            let keypair  = <AuthorityKeyPair as EncodeDecodeBase64>::decode_base64(&s);
+
+            Ok(AuthorityKeyPairWithPath::new(keypair.unwrap()))
+        }
+        serde_yaml::Value::Mapping(map) => {
+            let path = map.get(&Value::String("path".parse().unwrap())).ok_or_else(|| D::Error::custom("Missing path"))?;
+            println!("path: {:?}", path);
+            let keypair = read_authority_keypair_from_file(path.as_str().unwrap()).map_err(|e| D::Error::custom(format!("Failed to read keypair from file: {}", e)))?;
+            Ok(AuthorityKeyPairWithPath::new(keypair))
+        }
+        _ => {
+            Err(D::Error::custom("Invalid value type, expected string or mapping"))
+        }
+    }
+
+
+}
+
+
+## Rationale
+1.sui.keystore files comes from FileBasedKeystore::save , it should be aes-128 encode before save to files. also add decode step before new from files.
+2. for network.yaml,and fullnode.yaml, we should change the serder function to support both string and mapping type,
+and add aes-128 encode/decode step before save to files and new from files.
+
+
+
+## Backwards Compatibility
+
+There are no issues with backwards compatability.
```

Comment:
if you change the format of a config file, there are compatibility issue that some older sui binary may expect plain base64 but AES encrypted ones are included.

please discuss whether this feature is optional and what error or handling should the user see if there is incompatibility.